ports/stm32/rfcore.c

```diff
@@ -589,18 +600,34 @@ static const struct {
 void rfcore_ble_init(void) {
     DEBUG_printf("rfcore_ble_init\n");
 
-    // Clear any outstanding messages from ipcc_init.
-    tl_check_msg(&ipcc_mem_sys_queue, IPCC_CH_SYS, NULL);
-
     // Configure and reset the BLE controller.
-    tl_sys_hci_cmd_resp(HCI_OPCODE(OGF_VENDOR, OCF_BLE_INIT), (const uint8_t *)&ble_init_params, sizeof(ble_init_params), 0);
-    tl_ble_hci_cmd_resp(HCI_OPCODE(0x03, 0x0003), NULL, 0);
+    if (!rfcore_ble_reset()) {
+        // Out of standby initial init can fail. Reset rfcode and retry.
```

Comment:
I don't think this re-init-on-fail is needed anymore.  I tested without it and it seems OK.

But do you want to keep it in anyway, just in case?

Reply:
Yeah I was flipping back and forth on keeping it or not... I think it was really just to fix a specific problem that's already solved better now elsewhere. 
I say remove this retry; If there's ever a different problem that this _could_ workaround it's probably better we know it's happening and solve it properly too.

Reply:
https://github.com/micropython/micropython/pull/8337#discussion_r1127333834